File: Medium/0011-Container-With-Most-Water/Solution.rb
# @param {Integer[]} height
# @return {Integer}
def max_area(height)
    start = 0
    last = height.length - 1
    area = 0
    while start < last
        s = height[start]
        l = height[last]
        if (s <= l )&& (area < s * (last - start))
            area = s * (last - start)
        elsif (s >= l) && (area < l * (last - start))
            area = l * (last - start)
        end
        if s <= l
            start += 1
        elsif s >= l
            last -= 1
        end
    end
    area    
end

File: Medium/0011-Container-With-Most-Water/Solution.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    let start = 0;
  let last = height.length - 1;
  let area = 0;

  while (start < last) {
    const s = height[start];
    const l = height[last];
    const width = last - start;

    if (s <= l && area < s * width) {
      area = s * width;
    } else if (s >= l && area < l * width) {
      area = l * width;
    }

    if (s <= l) {
      start++;
    } else {
      last--;
    }
  }

  return area;
};